matRad_visApertureInfo.m

```diff
@@ -42,34 +42,53 @@ function matRad_visApertureInfo(apertureInfo,mode)
 color(:,3) = 0;
 color(:,2) = 0;
 
-% loop over all beams
-for i=1:numOfBeams
+if apertureInfo.runVMAT
+    % if doing VMAT, let wMax be the max weight across ALL angles
+    wMax = 0;
+    for i=1:numOfBeams
+        if wMax <= apertureInfo.beam(i).shape(1).weight
+            wMax = apertureInfo.beam(i).shape(1).weight;
+        end
+    end
+end
 
+
+for i=1:numOfBeams
+    
+    numOfShapes = numel(apertureInfo.beam(i).shape);
+    
     % open new figure for every beam
     figure('units','inches')
-
+    
     % get the MLC dimensions for this beam
     minX = apertureInfo.beam(i).MLCWindow(1);
-    maxX = apertureInfo.beam(i).MLCWindow(2);    
+    maxX = apertureInfo.beam(i).MLCWindow(2);
     
-    %get maximum weight
-    wMax = max([apertureInfo.beam(i).shape(:).weight]);
+    if ~apertureInfo.runVMAT
+        % if not VMAT, let wMax be the max weight of a particular angle
+        if numOfShapes;
+            wMax = max([apertureInfo.beam(i).shape(:).weight]);
+        end
+    end
     if strcmp(mode,'leafNum')
-
+        
         % get the active leaf Pairs
         % the leaf indices have to be flipped in order to fit to the order of
         % the leaf positions (1st row of leafPos is lowest row in physical
         % coordinates
         activeLeafInd = flipud(find(apertureInfo.beam(i).isActiveLeafPair));
     end
     
-    subplotColumns = ceil(apertureInfo.beam(i).numOfShapes/2);
-    subplotLines   = ceil(apertureInfo.beam(i).numOfShapes/subplotColumns);
+    %subplotColumns = ceil(apertureInfo.beam(i).numOfShapes/2);
+    %subplotLines   = ceil(apertureInfo.beam(i).numOfShapes/subplotColumns);
+    subplotColumns = ceil(numOfShapes/2);
+    subplotLines = ceil(numOfShapes/subplotColumns);
+    
     %adjust figure position
     set(gcf,'pos',[0 0 1.8*subplotColumns 3*subplotLines])
     
-    % loop over all shapes of the beam 
-    for j = 1:apertureInfo.beam(i).numOfShapes
+    % loop over all shapes of the beam
+    for j = 1:numOfShapes
```

Comment:
I am a little confused here about "numOfShapes". Because above, the numOfShapes is queried by counting the shape structs, which can be different from the "numOfShapes" property in apertureInfo.beam(i).numOfShapes. Was this always this way? @markbangert can you clarify?